### PR TITLE
add imageUpdateInterval

### PR DESCRIPTION
**Pull request by mcsdodo**
- make image update checking opt-in.
- split logic for container updates and image updates

### CODE REVIEW

**Comment by SiriosDev** [comment]
can i ask what do you mean with the distinction between container updates and image updates?

**Comment by mcsdodo** [comment]
> A Docker container is a self-contained, runnable software application or service. On the other hand, a Docker image is the template loaded onto the container to run it, like a set of instructions. 

I split logic of reporting state of running container (status/uptime/created etc.) and checking update of the image (newer version).

**Comment by SiriosDev** [comment]
> I split logic of reporting state of running container (status/uptime/created etc.) and checking update of the image (newer version).

in short terms every x minute check the "vitals" and every y minutes check "update"

**Comment by mcsdodo** [comment]
exactly

**Comment by MichelFR** [comment]
Thanks for your contribution!

I think we should probably rename some of the variables here, seems to be somewhat confusing.

ContainersMessages
ContainerMessage
ContainerMessages
ContainersMessage

We should also probably have a fallback if a user does not have a imageUpdateInterval set we set automatically take the default updateInterval so we dont have any breaking changes. We should also update the readme.md.

**Comment by mcsdodo** [comment]
done

File: src/services/HomeassistantService.ts [hunk 1]
```diff
@@ -420,7 +430,7 @@ export default class HomeassistantService {
    * @param container
    * @param client
    */
-  public static async publishDeviceMessage(container: ContainerInspectInfo, client: any) {
+  public static async publishContainerMessage(container: ContainerInspectInfo, client: any) {
```
[the remark this answers is not in the document]
Comment (mcsdodo) [reply src/services/HomeassistantService.ts:433]: No I don't think so. It 
1. parses container metadata (most of that should be cached somehow)
2. publishes to topic (either it's create or update - doesn't matter)

so basically these should be separate methods.

**Pull request merged**